src/fabric/src/fabric2_fdb.erl

```diff
@@ -985,18 +985,19 @@ get_fold_opts(RangePrefix, Options) ->
 
     % Set the maximum bounds for the start and endkey
     StartKey2 = case StartKey1 of
-        undefined -> <<>>;
-        SK2 -> SK2
+        undefined ->
+            <<RangePrefix/binary, 16#00>>;
+        SK2 ->
+            erlfdb_tuple:pack({SK2}, RangePrefix)
     end,
 
     EndKey2 = case EndKey1 of
-        undefined -> <<255>>;
-        EK2 -> EK2
+        undefined ->
+            <<RangePrefix/binary, 16#FF>>;
```

Comment:
It would be nice if we could switch to native FDB primitives here. Not sure how hard it would be.

Reply:
Hmm we already use `erlfdb_key:first_greater_than(StartKey2)`.

Reply:
I think we need to build this into the erlfdb api to be able to use. But I agree its a good idea.

Reply:
Narps. This is application logic, not fdb logic. first_greater_than gives you the first greater than the key. 16#FF on the end key is preventing the read from going outside the prefix defined range which would be perfectly valid for foundationdb but invalid for our required semantics.

These key ranges are fairly subtle. A lot of the various sorts and range limits were figured out by staring at the actual hex encodings of keys in fdb. This is mostly due to CouchDB having a much more complicated sorting behavior than strict byte comparisons but also with some surprising "Doh!" moments when figuring out when and when not to handle document ids in view rows and the like.